src/Components/test/E2ETest/ServerExecutionTests/ConnectionQualityTests.cs

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components.E2ETest.Infrastructure;
+using Microsoft.AspNetCore.Components.E2ETest.Infrastructure.ServerFixtures;
+using Microsoft.AspNetCore.E2ETesting;
+using OpenQA.Selenium;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.AspNetCore.Components.E2ETest.ServerExecutionTests
+{
+    public class ConnectionQualityTests : ServerTestBase<AspNetSiteServerFixture>
+    {
+        public ConnectionQualityTests(
+            BrowserFixture browserFixture,
+            AspNetSiteServerFixture serverFixture,
+            ITestOutputHelper output)
+            : base(browserFixture, serverFixture, output)
+        {
+            serverFixture.Environment = AspNetEnvironment.Development;
+            serverFixture.BuildWebHostMethod = TestServer.Program.BuildWebHost;
+        }
+
+        [Fact]
+        public void RefreshingThePageMultipleTimesCreatesNewCircuitsWithoutIssues()
+        {
+            Navigate("/connectionqualityhost");
+
+            var previousIdentifier = CaptureScopeIdentifier();
+            for (int i = 0; i < 50; i++)
+            {
+                Browser.Equal(i + 3, () => Browser.Manage().Cookies.AllCookies.Where(c => c.Name.StartsWith("Circuit.")).Count());
```

Comment:
Should there be a call to `Browser.Manage().Cookies.DeleteAllCookies();` before we get here? Otherwise it seems like this could be flaky.

Reply:
You're right, It was the only test in this class and then I ended up adding more.